### PR TITLE
Reset value of v-input to NULL if input is empty

### LINKED ISSUE #3450: Numeric fields can't be cleared
Optional numeric fields cannot be cleared after a value was set. Once saved they will be changed to 0 (but should be empty / NULL).

This is also true for internal numeric fields, like the min or max settings for said integer fields. Once I have set a minimum value for a numeric field I can no longer remove it and it will always reset to 0, thus requiring a minimum number of 0 for that input.

The app also doesn't differentiate between 0 and an empty input, as the save button of a collection item will stay disabled (as if the item is unchanged) if I try to delete a 0 from a numeric field.

### PR DESCRIPTION
**Pull request by nickrum**
Opening as a Draft for now to get some feedback before adjusting each and every `v-input`.

Fixes #3450.

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
Nice @nickrum! One thing that's worth discussing: what should it do with text-inputs? There is a difference between `''` and `null` and I'm not sure if we can / should say that all empty strings are saved as `null` 🤔  (though that would make sense for required I suppose, as an "empty" input shouldn't pass the required validation? 🤔 )

@benhaynes

**Comment by benhaynes** [comment]
Could be an interface option, whether to save "nothing" as NULL or empty string. Or could dynamically be based on if the field "allows null". Or could have options for both in the Field Label dropdown.

Not sure if we'll ever be able to _assume_ which one to do... so I don't have any good answer here.

**Comment by rijkvanzanten** [comment]
For numeric it's an easier answer, as there is no `''` in the first place, and `0` is clearly a value with `null` being "empty". I agree that we should add it as an option to the `text-input` interface to set "empty = null"

**Comment by nickrum** [comment]
I agree that there should be an option. From a users perspective there is no visual difference between `''` and `NULL`, so it is likely a good idea to not mix those within a single field.

File: app/src/lang/translations/en-US.yaml [hunk 1]
```diff
@@ -922,6 +922,8 @@ interfaces:
     trim_label: Trim the start and end
     mask: Masked
     mask_label: Hide the real value
+    nullable: Nullable
+    nullable_label: Send NULL as empty
```
Comment (rijkvanzanten) [inline app/src/lang/translations/en-US.yaml:926]:
```suggestion
    nullable_label: Save empty value as NULL
```

**Review by rijkvanzanten — DISMISSED** [review]

**Comment by rijkvanzanten** [comment]
@nickrum Implementation LGTM! By defaulting `nullable` to true, we'll have to make sure it doesn't have any side effects elsewhere in the app though 🤔

**Comment by nickrum** [comment]
@rijkvanzanten The reason for defaulting `nullable` to true was that the default value of `value` is `null`.

I already started checking every `v-input` for regressions, but this might take a while.

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**